docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,296 @@
-      update (more on this later).
-    * Updaters expected to modify the Git working tree in-place
+Much of the time, it will only be necessary to supply one command to
+be run.
 
-    After invoking the updaters, `fluxd` will then commit and push the resulting modifications to the git repository.
+The commands will be run with the target path being processed as a
+working directory -- which is not necessarily the same directory in
+which the `.flux.yaml` file was found. [See below](#execution-context)
+for more details on the execution context in which commands are run.
 
-3. `fluxd` will ignore any other yaml files under that path (e.g. resource manifests).
+### Using patch-updated configuration
 
-Generators and updaters are intentionally independent, in case a matching updater cannot be provided. It is hard to
-create updaters for some factorization technologies (particularly Configuration-As-Code). To improve the situation,
-a separate configuration file variant (`patchedUpdated`) is provided, which will be described later on.
+A patch-updated configuration generates manifests using commands, and
+records updates as a set of [strategic merge][strategic-merge] patches
+in a file.
 
-### Execution context of commands
+For example, when an automated image upgrade is run, fluxd will do
+this:
 
-`generators` and `updaters` are run in a POSIX shell inside the Flux container. This means that the `command`s supplied
-should be available in the [Flux container image](https://github.com/fluxcd/flux/blob/master/docker/Dockerfile.flux).
-Flux currently includes `Kustomize` and basic Unix shell tools. If the tools in the Flux image are not sufficient for
-your use case, you can include new tools in your own Flux-based image or, if the tools are popular enough, Flux
-maintainers can add them to the Flux image (please create an issue). In the future (once [Ephemeral
-containers](https://github.com/kubernetes/kubernetes/pull/59416) are available), you will be able to specify an container
-image for each command.
+ 1. run the generator commands and parse the manifests;
+ 2. find the manifest that needs to be updated, and calculate the
+    patch to it that performs the update;
+ 3. record that patch in the patch file.
 
-The working directory (also known as CWD) of the `command`s executed from a `.flux.yaml` file will be set to the
-target path (`--git-path` entry) used when finding that `.flux.yaml` file.
+When syncing, fluxd will generate the manifests as usual, then apply
+all the patches that have been recorded in the patch file.
 
-For example, when using flux with `--git-path=staging` on a git repository with this structure:
+This is how a patch-updated `.flux.yaml` looks:
 
-```sh
-├── .flux.yaml
-├── staging/
-├──── [...]
-├── production/
-└──── [...]
+```yaml
+version: 1
+patchUpdated:
+  generators:
+  - command: jk generate ./index.js
```

Comment:
It may be worth indicating that `jk` is not in the container, or indicate how to install it, or simply use `kustomize`. Otherwise I think it may be confusing.

Reply:
I suppooooooose

Reply:
Changed it to be `generator_command`, so as not to distract from the shape of the YAML, which is the point here.